src/lightning_lite/strategies/strategy.py

```diff
@@ -0,0 +1,298 @@
+
+    def setup_environment(self) -> None:
+        """Setup any processes or distributed connections.
+
+        This must be called by the framework at the beginning of every process, before any distributed communication
+        takes place.
+        """
+        assert self.accelerator is not None
+        self.accelerator.setup_device(self.root_device)
+
+    def process_dataloader(self, dataloader: DataLoader) -> DataLoader:
+        """Wraps the dataloader if necessary.
+
+        Args:
+            dataloader: iterable. Ideally of type: :class:`torch.utils.data.DataLoader`
+        """
+        return dataloader
+
+    def setup_module_and_optimizers(
+        self, module: Module, optimizers: List[Optimizer]
+    ) -> Tuple[Module, List[Optimizer]]:
+        """Set up a model and multiple optimizers together.
+
+        The returned objects are expected to be in the same order they were passed in. The default implementation will
+        call :meth:`_setup_model` and :meth:`_setup_optimizer` on the inputs.
+        """
+        module = self.setup_module(module)
+        optimizers = [self.setup_optimizer(optimizer) for optimizer in optimizers]
+        return module, optimizers
+
+    def setup_module(self, module: Module) -> Module:
+        """Performs setup for the model, e.g., by wrapping it by another class."""
+        return module
+
+    def setup_optimizer(self, optimizer: Optimizer) -> Optimizer:
+        """Performs setup for the optimizer, e.g., by wrapping it by another class."""
+        return optimizer
+
+    @abstractmethod
+    def module_to_device(self, module: Module) -> None:
+        """Moves the model to the correct device."""
+
+    def batch_to_device(self, batch: Any, device: Optional[torch.device] = None) -> Any:
+        """Moves the batch to the correct device.
+
+        The returned batch is of the same type as the input batch, just
+        having all tensors on the correct device.
+
+        Args:
+            batch: The batch of samples to move to the correct device
+            device: The target device
+        """
+        device = device or self.root_device
+        return move_data_to_device(batch, device)
+
+    @contextlib.contextmanager
+    def module_sharded_context(self) -> Generator:
+        """Provide hook to create modules in a distributed aware context. This is useful for when we'd like to
+        shard the model instantly, which is useful for extremely large models which can save memory and
+        initialization time.
+
+        Returns: Model parallel context.
+        """
+        yield
+
+    def pre_backward(self, tensor: Tensor, module: Optional[Module]) -> None:
+        """Run before precision plugin executes backward."""
+
+    def backward(self, tensor: Tensor, module: Optional[Module], *args: Any, **kwargs: Any) -> None:
+        r"""Forwards backward-calls to the precision plugin."""
+        self.pre_backward(tensor, module)
```

Comment:
I thought you had no use for pre and post in Lite?

Reply:
Hmm, there are some strategies that implement, but it's not relevant for Lite atm. 
It's probably better to remove it for now yes.